--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -7,7 +7,7 @@
 
 /*----------------------------------------------------------------------------*/
 #include <linux/module.h>
-#include <linux/input-polldev.h>
+#include <linux/input.h>
 #include <linux/platform_device.h>
 #include <linux/iio/consumer.h>
 #include <linux/version.h>
@@ -229,7 +229,7 @@
 
 };
 
-extern struct input_dev * joypad_input_g;
+
 
 static int pwm_vibrator_start(struct joypad *joypad)
 {
@@ -384,9 +384,9 @@
 };
 
 /*----------------------------------------------------------------------------*/
-static void joypad_gpio_check(struct input_polled_dev *poll_dev)
-{
-	struct joypad *joypad = poll_dev->private;
+static void joypad_gpio_check(struct input_dev *poll_dev)
+{
+	struct joypad *joypad = input_get_drvdata(poll_dev);
 	int nbtn, value;
 
 	for (nbtn = 0; nbtn < joypad->bt_gpio_count; nbtn++) {
@@ -394,20 +394,20 @@
 
 		value = gpiod_get_value_cansleep(gpio->desc);
 		if (value != gpio->old_value) {
-			input_event(poll_dev->input,
+			input_event(poll_dev,
 				gpio->report_type,
 				gpio->linux_code,
 				value == 1);
 			gpio->old_value = value;
 		}
 	}
-	input_sync(poll_dev->input);
-}
-
-/*----------------------------------------------------------------------------*/
-static void joypad_adc_check(struct input_polled_dev *poll_dev)
-{
-	struct joypad *joypad = poll_dev->private;
+	input_sync(poll_dev);
+}
+
+/*----------------------------------------------------------------------------*/
+static void joypad_adc_check(struct input_dev *poll_dev)
+{
+	struct joypad *joypad = input_get_drvdata(poll_dev);
 	int nbtn;
 	int mag;
 
@@ -465,20 +465,20 @@
 		adcy->value = adcy->value > adcy->max ? adcy->max : adcy->value;
 		adcy->value = adcy->value < adcy->min ? adcy->min : adcy->value;
 
-		input_report_abs(poll_dev->input,
+		input_report_abs(poll_dev,
 			adcx->report_type,
 			adcx->invert ? adcx->value * (-1) : adcx->value);
-		input_report_abs(poll_dev->input,
+		input_report_abs(poll_dev,
 			adcy->report_type,
 			adcy->invert ? adcy->value * (-1) : adcy->value);
 	}
-	input_sync(poll_dev->input);
-}
-
-/*----------------------------------------------------------------------------*/
-static void joypad_poll(struct input_polled_dev *poll_dev)
-{
-	struct joypad *joypad = poll_dev->private;
+	input_sync(poll_dev);
+}
+
+/*----------------------------------------------------------------------------*/
+static void joypad_poll(struct input_dev *poll_dev)
+{
+	struct joypad *joypad = input_get_drvdata(poll_dev);
 
 	if (joypad->enable) {
 		if (joypad->use_miyoo_serial) {
@@ -487,11 +487,11 @@
 			 * our miyoo_serial_threadfn(). We'll just do
 			 * input_report_abs here from the stored values.
 			 */
-			input_report_abs(poll_dev->input, ABS_X,  joypad->miyoo.left_x);
-			input_report_abs(poll_dev->input, ABS_Y,  joypad->miyoo.left_y);
-			input_report_abs(poll_dev->input, ABS_RX, joypad->miyoo.right_x);
-			input_report_abs(poll_dev->input, ABS_RY, joypad->miyoo.right_y);
-			input_sync(poll_dev->input);
+			input_report_abs(poll_dev, ABS_X,  joypad->miyoo.left_x);
+			input_report_abs(poll_dev, ABS_Y,  joypad->miyoo.left_y);
+			input_report_abs(poll_dev, ABS_RX, joypad->miyoo.right_x);
+			input_report_abs(poll_dev, ABS_RY, joypad->miyoo.right_y);
+			input_sync(poll_dev);
 
 			joypad_gpio_check(poll_dev);
 		} else {
@@ -499,17 +499,17 @@
 			joypad_gpio_check(poll_dev);
 		}
 	}
-	if (poll_dev->poll_interval != joypad->poll_interval) {
+	if (input_get_poll_interval(poll_dev) != joypad->poll_interval) {
 		mutex_lock(&joypad->lock);
-		poll_dev->poll_interval = joypad->poll_interval;
+		input_set_poll_interval(poll_dev, joypad->poll_interval);
 		mutex_unlock(&joypad->lock);
 	}
 }
 
 /*----------------------------------------------------------------------------*/
-static void joypad_open(struct input_polled_dev *poll_dev)
-{
-	struct joypad *joypad = poll_dev->private;
+static int joypad_open(struct input_dev *poll_dev)
+{
+	struct joypad *joypad = input_get_drvdata(poll_dev);
 	int nbtn;
 
 	for (nbtn = 0; nbtn < joypad->bt_gpio_count; nbtn++) {
@@ -520,11 +520,11 @@
 		gpio->old_value = val;
 
 		// Immediately report the current state
-		input_event(poll_dev->input, gpio->report_type,
+		input_event(poll_dev, gpio->report_type,
 					gpio->linux_code,
 					val == 1);
 	}
-	input_sync(poll_dev->input);
+	input_sync(poll_dev);
 
 	for (nbtn = 0; nbtn < joypad->amux_count; nbtn++) {
 		struct bt_adc *adc = &joypad->adcs[nbtn];
@@ -544,12 +544,13 @@
 	mutex_lock(&joypad->lock);
 	joypad->enable = true;
 	mutex_unlock(&joypad->lock);
-}
-
-/*----------------------------------------------------------------------------*/
-static void joypad_close(struct input_polled_dev *poll_dev)
-{
-	struct joypad *joypad = poll_dev->private;
+	return 0;
+}
+
+/*----------------------------------------------------------------------------*/
+static void joypad_close(struct input_dev *poll_dev)
+{
+	struct joypad *joypad = input_get_drvdata(poll_dev);
 
 	if (joypad->has_rumble) {
 		cancel_work_sync(&joypad->play_work);
@@ -832,7 +833,7 @@
 /*----------------------------------------------------------------------------*/
 static int joypad_input_setup(struct device *dev, struct joypad *joypad)
 {
-	struct input_polled_dev *poll_dev;
+	struct input_dev *poll_dev;
 	struct input_dev *input;
 	int nbtn, error;
 	u32 joypad_bustype = BUS_HOST;
@@ -840,23 +841,25 @@
 	u32 joypad_revision = 0;
 	u32 joypad_product = 0;
 
-	poll_dev = devm_input_allocate_polled_device(dev);
+	poll_dev = devm_input_allocate_device(dev);
 	if (!poll_dev) {
 		dev_err(dev, "no memory for polled device\n");
 		return -ENOMEM;
 	}
 
-	poll_dev->private	= joypad;
-	poll_dev->poll		= joypad_poll;
-	poll_dev->poll_interval	= joypad->poll_interval;
+	input_set_drvdata(poll_dev, joypad);
+	error = input_setup_polling(poll_dev, joypad_poll);
+	if (error)
+		return error;
+	input_set_poll_interval(poll_dev, joypad->poll_interval);
 	poll_dev->open		= joypad_open;
 	poll_dev->close		= joypad_close;
 
-	input = poll_dev->input;
+	input = poll_dev;
 
 	input->name = DRV_NAME;
 
-	joypad_input_g=input;
+
 
 	device_property_read_string(dev, "joypad-name", &input->name);
 	input->phys = DRV_NAME"/input0";
@@ -943,7 +946,7 @@
 
 	joypad->dev = dev;
 
-	error = input_register_polled_device(poll_dev);
+	error = input_register_device(poll_dev);
 	if (error) {
 		dev_err(dev, "unable to register polled device, err=%d\n",
 			error);
